documentation-website/Writerside/topics/Deep-Dive-into-DSL.md

```diff
@@ -766,3 +766,65 @@ The values specified in the statement block will be used for the insert statemen
 In the example above, if the original row was inserted with a user-defined <code>rating</code>, then <code>replace()</code> was executed with a block that omitted the <code>rating</code> column, 
 the newly inserted row would store the default rating value. This is because the old row was completely deleted first.
 </note>
+
+## Column transformation
+
+Column transformations allow to define custom transformations between database column types and application's data types. 
+This can be particularly useful when you need to store data in one format but work with it in another format within your application.
+
+Consider the following example, where we define a table to store meal times and transform these times into meal types:
+
+```kotlin
+enum class Meal {
+    BREAKFAST,
+    LUNCH,
+    DINNER
+}
+
+object Meals : Table() {
+    val mealTime = time("meal_time")
+        .transform(
+            toReal = {
+                when {
+                    it.hour < 10 -> Meal.BREAKFAST
+                    it.hour < 15 -> Meal.LUNCH
+                    else -> Meal.DINNER
+                }
+            },
+            toColumn = {
+                when (it) {
+                    Meal.BREAKFAST -> LocalTime(8, 0)
+                    Meal.LUNCH -> LocalTime(12, 0)
+                    Meal.DINNER -> LocalTime(18, 0)
+                }
+            }
+        )
+}
+```
+
+The `transform` function is used to apply custom transformations to the `mealTime` column:
+
+- The `toReal` function transforms the stored `LocalTime` values into `Meal` enums. It checks the hour of the stored time and returns the corresponding meal type.
+- The `toColumn` function transforms `Meal` enums back into `LocalTime` values for storage in the database.
+
+Transformation could be also defined as an implementation of `ColumnTransformer` interface and reused among different tables:
+
+```kotlin
+class MealTimeTransformer : ColumnTransformer<Meal, LocalTime> {
+    override fun fromColumn(value: LocalTime): Meal = when {
```

Comment:
Name should be `toReal()`. I don't mind either, but I think it'd be nice if the parameter names in `transform()` match the function names in `ColumnTransformer`.